### DISCUSSION TITLE
Ideas about supporting fractional rack units and how we might solve it

### DISCUSSION THREAD

**Discussion by jathanism** [nautobot/nautobot#414] [Ideas]
There have been some inquiries about whether Nautobot can support fractional rack units. It can't today. But what if? 

Valid use cases
- Blade chassis
- Devices that are the same unit height but don't fill width or depth of the rack elevation/shelf

Current potential ways to implement it
- Using a parent device and attaching child devices using device bays (a hack if this doesn't model the real world)
- Or model a “shelf” device and track the actual devices as un-racked, but still assigned to the rack (another hack)

Challenges
- Drawing rack elevation diagrams with fractional height/widths
- Calculating available space for fractional units

**Comment by ancker010** [comment]
I think it might be helpful to specify the concern is rack width, not rack unit.
A 1U shelf with two mac minis, netbeez, or something else is still 1U, but each device would be 1/2 width, etc.

**Reply by jathanism** [reply]
That sounds like your ask, so thanks for clarifying, but there's also some internal discussion about fractional heights, like 1.5U, which also is not currently possible.

**Reply by ancker010** [reply]
1.5U is blasphemy. :)
